### ISSUE TITLE
Memory leak? 

### ISSUE THREAD

**Issue by QiAnXinCodeSafe** [opencv/opencv#26576]
Hi all,
This is Qianxin CodeSafe Team, we found a suspicious issue, at
https://github.com/opencv/opencv/blob/23fcea0d33b1a17525ce84b31400d409ccc14b6a/3rdparty/quirc/src/quirc.c#L81
if the value of “pixels” isn't null at
https://github.com/opencv/opencv/blob/23fcea0d33b1a17525ce84b31400d409ccc14b6a/3rdparty/quirc/src/quirc.c#L82
and if conditional value is false at
https://github.com/opencv/opencv/blob/23fcea0d33b1a17525ce84b31400d409ccc14b6a/3rdparty/quirc/src/quirc.c#L91
To sum up,this will result in a  memory leak of point “pixels”

**Comment by opencv-alalek** [comment]
Looks like this 3rdparty function (`quirc_resize`) is not used in OpenCV (not called directly or indirectly).

If you want a fix for this issue please report to upstream repository.

---

> if the value of “pixels” is null

then line 91 is not executed at all due to "goto" statement (to the line 98).

> and if conditional value is false at

Checks on the line 91 and on the 80 (precondition for 81) are the same. They could not be true and false.

**Comment by Kumataro** [comment]
In addition, `QUIRC_PIXEL_ALIAS_IMAGE` is defined following code. 
So we can consider this condition will not be changed after compiling.

https://github.com/opencv/opencv/blob/23fcea0d33b1a17525ce84b31400d409ccc14b6a/3rdparty/quirc/include/quirc_internal.h#L26-L42

**Comment by QiAnXinCodeSafe** [comment]
Assume that the value of "pixels" is non-empty at line 82 and and  conditional value is false at  line 91,this maybe result in a  a memory leak of point “pixels”?

**Comment by asmorkalov** [comment]
OpenCV no not use Quirc library by default since 4.9.0. Please submit a bug or patch to upstream repository.

**Comment by Kumataro** [comment]
upstream https://github.com/dlbeer/quirc

But your point may be false-positive.
Quirc source code is according to the conditions and `pixels` value below. 
When you consider `quirc_destroy()` function is called after calling `quirc_resize()` function, `pixels` will be released at all confition. So there are no problem.

If the problem is still unresolved, could you please clarify under what conditions it occurs?

[base]  Extract only the part related to pixels
-------------------------------------------------

```
int quirc_resize(struct quirc *q, int w, int h)
{
	quirc_pixel_t	*pixels = NULL;

	if (!QUIRC_PIXEL_ALIAS_IMAGE) {
		pixels = calloc(newdim, sizeof(quirc_pixel_t));
		if (!pixels)
			goto fail;
	}

	if (!QUIRC_PIXEL_ALIAS_IMAGE) {
		free(q->pixels);
		q->pixels = pixels;
	}

	return 0;
	/* NOTREACHED */
fail:
	free(pixels);

	return -1;
}

void quirc_destroy(struct quirc *q)
{
	if (!QUIRC_PIXEL_ALIAS_IMAGE)
		free(q->pixels);
}
```

[case 1] If `!QUIRC_PIXEL_ALIAS_IMAGE` is true
------------------------------------------------

```
int quirc_resize(struct quirc *q, int w, int h)
{
	quirc_pixel_t	*pixels = NULL;

	pixels = calloc(newdim, sizeof(quirc_pixel_t));
	if (!pixels)
		goto fail;

	free(q->pixels);
	q->pixels = pixels;

	return 0;
	/* NOTREACHED */
fail:
	free(pixels); // free(NULL) works nothing.

	return -1;
}

void quirc_destroy(struct quirc *q)
{
	free(q->pixels);
}
```

[case 2] If `!QUIRC_PIXEL_ALIAS_IMAGE` is false.
------------------------------------------------

```
int quirc_resize(struct quirc *q, int w, int h)
{
	quirc_pixel_t	*pixels = NULL;

	return 0;
	/* NOTREACHED */
fail:
	free(pixels);

	return -1;
}

void quirc_destroy(struct quirc *q)
{
}

```
